File: Backend/Accounting101.Ledger.Api/Control/CapabilitySet.cs
using MongoDB.Bson.Serialization.Attributes;

namespace Accounting101.Ledger.Api.Control;

/// <summary>
/// A named, editable bundle of capabilities in the deployment's control database. Sets are the
/// owner-managed successor to the hardcoded <see cref="RolePresets"/>: members reference sets and
/// resolve their capabilities from them (AC-2). Deployment-wide — one catalog per deployment.
/// </summary>
[BsonIgnoreExtraElements]
public sealed class CapabilitySet
{
    [BsonId]
    public Guid Id { get; set; }

    /// <summary>Unique, human-facing name (e.g. "Controller", "Warehouse Clerk").</summary>
    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    /// <summary>The capabilities this set grants — each a member of <see cref="Capabilities.All"/>.</summary>
    public IReadOnlyList<string> Capabilities { get; set; } = [];

    /// <summary>True for sets seeded from <see cref="RolePresets"/>: editable in place, but not deletable.</summary>
    public bool Builtin { get; set; }
}

File: UI/Angular/src/app/core/capability-sets/capability-set.ts
export interface CapabilitySet {
  id: string;
  name: string;
  description?: string;
  capabilities: string[];
  builtin: boolean;
  affectedMemberCount: number;
  restricted: boolean;
}

export interface CreateCapabilitySetRequest {
  name: string;
  description?: string;
  capabilities: string[];
  restricted?: boolean;
}

export interface UpdateCapabilitySetRequest {
  name: string;
  description?: string;
  capabilities: string[];
  restricted?: boolean;
}
